File: Leetcode/Hash Table/#349 Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
     unordered_set<int>st;
        vector<int>res;
        for(auto i: nums1)
            st.insert(i);
        for(auto i: nums2){
            if(st.count(i)){
                res.push_back(i);
                st.erase(i);
            }
        }
        return res;
    }
};

File: Array/#349. Intersection of Two Arrays.py
'''
TC: O(n)
SC: O(n)
'''

class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        visit = set()
        res = []
        i, j = 0, 0
        for e in nums1:
            visit.add(e)
        for e in nums2:
            if e in visit:
                res.append(e)
                visit.remove(e)

        return res